app/controllers/submit.js

```diff
@@ -899,11 +1019,25 @@ export default Ember.Controller.extend(Analytics, BasicsValidations, NodeActions
             const node = this.get('node');
             this.set('savingPreprint', true);
             this.toggleProperty('shareButtonDisabled');
-            model.set('isPublished', true);
+
+            let log = null;
+            if (this.get('moderationType')) {
+                log = this.get('store').createRecord('reviewLog', {
+                   action: 'submit',
+                   reviewable: this.get('model')
+                });
+            } else {
+                model.set('isPublished', true);
+            }
             node.set('public', true);
 
             return model.save()
-                .then(() => node.save())
+                .then(() => {
+                    node.save();
+                    if (log) {
```

Comment:
You should be chaining the `.save()`s here to ensure they resolve successfully.
 If you don't care about the resolution order, you can throw them all into a [`RVSP.all`](https://www.emberjs.com/api/ember/2.14/classes/RSVP/methods/all?anchor=all)